### PR TITLE
[SPARK-50140][BUILD] Upgrade `zstd-jni` to 1.5.6-7

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to upgrade `zstd-jni` from `1.5.6-6` to `1.5.6-7`.

### Why are the changes needed?
1.v1.5.6-6 VS v1.5.6-7
https://github.com/luben/zstd-jni/compare/v1.5.6-6...v1.5.6-7
https://github.com/luben/zstd-jni/commit/0c2051bef1a86162ed68e3781828faaefa34eb1c

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -836,7 +836,7 @@
       <dependency>
         <groupId>com.github.luben</groupId>
         <artifactId>zstd-jni</artifactId>
-        <version>1.5.6-6</version>
+        <version>1.5.6-7</version>
```
Comment (dongjoon-hyun) [inline pom.xml:839]: Is this released without tagging? It seems that the last tag is still `1.5.6-6`.
- https://github.com/luben/zstd-jni/tags

Comment (dongjoon-hyun) [reply pom.xml:839]: Maybe, is this not announced officially yet?

Comment (panbingkun) [reply pom.xml:839]: Yes, I think it's just that it hasn't been announced officially yet. But its GitHub has already been marked as follows:
https://github.com/luben/zstd-jni
image

**Comment by dongjoon-hyun** [comment]
Let's wait for a few days until the announcement comes out.

**Comment by panbingkun** [comment]
> Let's wait for a few days until the announcement comes out.

Don't hurry, let's wait for it 😄

**Comment by dongjoon-hyun** [comment]
Could you ping the author of `zstd-jni` to get the confirmation?

**Comment by dongjoon-hyun** [comment]
Never mind. I filed an issue and pinged you.
- https://github.com/luben/zstd-jni/issues/329

**Comment by panbingkun** [comment]
Hi @luben, sorry to bother you
Can you help confirm if the `1.5.6-7` version of  `zstd-jni` is officially released?
- Because I can observe the following information:
https://mvnrepository.com/artifact/com.github.luben/zstd-jni/1.5.6-7
image

- However, it does not have a tag description:
https://github.com/luben/zstd-jni/tags

Thank you very much! ❤️

**Comment by panbingkun** [comment]
> Never mind. I filed an issue and pinged you.
> 
> * [v1.5.6-7 is released officially? luben/zstd-jni#329](https://github.com/luben/zstd-jni/issues/329)

Thanks ! ❤️

**Comment by luben** [comment]
yes it's officially released, I just forgot to `git push --tags`

**Comment by dongjoon-hyun** [comment]
Thank you for the confirmation, @luben .

Then, merged to master for Apache Spark 4.0.0 on February 2025.

**Pull request closed without merging**